pkgs/applications/editors/lapce/default.nix

```diff
@@ -17,28 +17,33 @@
 , ApplicationServices
 , Carbon
 , AppKit
+, wrapGAppsHook
+, gobject-introspection
 }:
-
 rustPlatform.buildRustPackage rec {
   pname = "lapce";
-  version = "0.1.2";
+  version = "0.2.0";
 
   src = fetchFromGitHub {
     owner = "lapce";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-jH473FdBI3rGt90L3WwMDPP8M3w0rtG5D758ceCMw94=";
+    sha256 = "sha256-cCcI5V6CMLkJM0miLv/o7LAJedrgb+z2CtWmF5/dmvY=";
   };
 
-  cargoSha256 = "sha256-0Kya2KcyBDlt0TpFV60VAA3+JPfwId/+k8k+H97EhB0=";
+  cargoSha256 = "sha256-4njr6W42DnBItmz6Yrmu5pCR13/Bfv10BxqgoP4JKS0=";
 
   nativeBuildInputs = [
     cmake
     pkg-config
     perl
     copyDesktopItems
+    wrapGAppsHook # FIX: No GSettings schemas are installed on the system
+    gobject-introspection
   ];
 
+  cargoPatches = [ ./treesitterfix.patch ];
```

Comment:
This is from https://github.com/lapce/lapce/compare/v0.2.0...MinusGix:36c0eb50616263ea96d3535dc0be2432f1888b88 right?

Then first for the size we should fetchpatch it and alternative just fetch the right source straight away and then apply the one patch we need.

eg fetch https://github.com/lapce/lapce/commit/c5a924ef34250e9117e2b57c19c1f29f6b9b3ea7 and https://github.com/lapce/lapce/commit/36c0eb50616263ea96d3535dc0be2432f1888b88.patch

Reply:
> eg fetch https://github.com/lapce/lapce/commit/c5a924ef34250e9117e2b57c19c1f29f6b9b3ea7 and https://github.com/lapce/lapce/commit/36c0eb50616263ea96d3535dc0be2432f1888b88.patch

So instead of using v0.2.0, should I just use https://github.com/lapce/lapce/commit/c5a924ef34250e9117e2b57c19c1f29f6b9b3ea7 and then patch it through https://github.com/lapce/lapce/commit/36c0eb50616263ea96d3535dc0be2432f1888b88.patch? Just checked, this patch is already in https://github.com/lapce/lapce/commit/c5a924ef34250e9117e2b57c19c1f29f6b9b3ea7.

But what about the `version` attribute? Right now it's `v0.2.0`, but what happened if I just use the currently latest master commit (at least that's what I got from your explanation).

Reply:
@SuperSandro2000 So it's preferred to use fetchpatch to get the patch from the magic github link to keep repo size smaller?

Reply:
Just [answered](https://nixos.org/manual/nixpkgs/unstable/#sec-patches) my own question, "In the interest of keeping our maintenance burden to a minimum, only patches that are unique to nixpkgs should be added in this way."

Reply:
> So instead of using v0.2.0, should I just use https://github.com/lapce/lapce/commit/c5a924ef34250e9117e2b57c19c1f29f6b9b3ea7 and then patch it with https://github.com/lapce/lapce/commit/36c0eb50616263ea96d3535dc0be2432f1888b88.patch? Just checked, this patch is already in https://github.com/lapce/lapce/commit/c5a924ef34250e9117e2b57c19c1f29f6b9b3ea7.

@0x715C can I also ask what do you think of this? Am I understanding what @SuperSandro2000 right?

Reply:
@tengkuizdihar Unsure where `https://github.com/lapce/lapce/commit/c5a924ef34250e9117e2b57c19c1f29f6b9b3ea7` comes into it, 0.2.0 is the tagged release that we want. As for getting the treesitter patch, I think Sandro means something like this.
```
treesitterpatch = fetchpatch {
  url = https://github.com/lapce/lapce/compare/v0.2.0...MinusGix:36c0eb5.diff;
  sha256 = "sha256-mvTrRzh/Pnys26Qj5BPglFX4A3HOy/CqhwpxVt7PyhA=";
};

cargoHash = "sha256-4njr6W42DnBItmz6Yrmu5pCR13/Bfv10BxqgoP4JKS0=";

cargoPatches = [ treesitterpatch ];
```

Reply:
Sure, I will go with that for now

Reply:
Donee

Reply:
> Just checked, this patch is already in [lapce/lapce@c5a924e](https://github.com/lapce/lapce/commit/c5a924ef34250e9117e2b57c19c1f29f6b9b3ea7).

Then we should use that straight away which makes it way more clearer than downloading a big patch.

> But what about the `version` attribute? Right now it's `v0.2.0`, but what happened if I just use the currently latest master commit (at least that's what I got from your explanation).

It is going to contain the date which is ``unstable-2022-09-21`` for the linked commit.

> @SuperSandro2000 So it's preferred to use fetchpatch to get the patch from the magic github link to keep repo size smaller?

Yes, especially if the patch is 7k lines or so. Also the link is not magic.



> url = "https://github.com/lapce/lapce/compare/v0.2.0...MinusGix:36c0eb5.diff[](https://github.com/lapce/lapce/compare/v0.2.0...MinusGix:36c0eb5.diff)";

Using short shas is forbidden because they can become invalid over time. Also using links from forks is not that nice because they could be deleted in the future and then the link is no longer working.

----

We should switch to downloading an unstable version from the commit because we are essentially doing that already, only more complicated. If there is a new release with the fixes we need we can switch to that.

Reply:
done again! @SuperSandro2000 I'm using unstable now without any patches.

Reply:
Bumping again @SuperSandro2000 (if it's good please tell me how to merge it :pray:)

Reply:
If it's good, then he'll merge it. Usually only people who have write access to Nixpkgs can merge it, but anyone (?) can review.

Reply:
Bumping again @SuperSandro2000, I've got changes from yesterday according to your comments